### ISSUE TITLE
Typescript Support: Doesn't re-interperet typescript imports as they change. Must restart Svelte server every time.

### ISSUE THREAD

**Issue by samdesota** [sveltejs/language-tools#177]
**Describe the bug**
Thanks for investing time on this, auto imports alone are massively valuable with svelte.

Unfortunately, as I continue to change typescript files, the svelte imports don't update. If I add a new function or update a functions type in a typescript file outside the svelte file, svelte doesn't recognize it, event when I save the changes. 

To get svelte to update, I restart the language server.

**To Reproduce**
Create project, tsconfig / svelte config below. Use latest version of the VSCode. Observe language server does not observe changes from imported typescript files.

tsconfig:
```
{
  "compilerOptions": {
    "module": "commonjs",
    "downlevelIteration": true,
    "noImplicitAny": false,
    "removeComments": true,
    "preserveConstEnums": true,
    "sourceMap": true,
    "baseUrl": "src",
    "target": "ES2019",
    "esModuleInterop": true,
    "isolatedModules": true,
    "lib": ["ES2019", "dom"],
    "types": ["svelte", "@sapper", "global"],
    "typeRoots": ["typings"]
  },

  "include": ["src/**/*"]
}
```

svelte.config.js
```
const preprocess = require("svelte-preprocess");

module.exports = {
  preprocess: [preprocess],
};
```
**System (please complete the following information):**
 - OS: macOS 10.15.4
 - IDE: VSCode
 - Plugin/Package: Svelte Plugin v. 0.9.3

**Comment by jasonlyu123** [comment]
[svelte beta](https://marketplace.visualstudio.com/items?itemName=svelte.svelte-vscode) has this now.

**Comment by samdesota** [comment]
Ah, got it. They looked like the same extension. I guess I'll close this if it's on the pipeline, using Svelte Beta now and it's working well. Thank you!

**Comment by samdesota** [comment]
Hmm, after some more testing I'm still not seeing updates propagate into svelte modules.

**Comment by jasonlyu123** [comment]
There's a known issue that the diagnostics don't get updated if dependent file changes, other things like completion or hover should be fine. You could leave this open for diagnostics.

**Comment by dummdidumm** [comment]
If it's about typescript files you don't see changes of: you have to a) save the files (editing is not enough) and b) if it's about diagnostics, edit the svelte file (like put a space in or sth). Reason: the language server only triggers on svelte files. Typescript file updates are only propagated through a file watcher and that needs to see the changes on disk, not just in the editor.

**Comment by jasonlyu123** [comment]
About diagnostics, there's some workaround. 
https://github.com/vuejs/vetur/blob/5f9974f3cd694f64518fb5d362d60c6370ec52bf/server/src/services/vls.ts#L208

I tried this [before](https://github.com/jasonlyu123/language-tools/commit/54d9619e3bd138fe69339534f59945bb030361a6) but never make a PR for it. To get better performance we could change it to only do the full update when 
`WillSaveTextDocument` on `svelte` file and `DidChangeWatchedFiles` on `js/ts` file

**Comment by dummdidumm** [comment]
@samdesota given the info we provided about the current limitations, does the issue still occur after using our workarounds?

**Comment by peterreeves** [comment]
Hi. I've been chasing what sounds like this problem all day. I started with this repo:
https://github.com/codechips/svelte-and-typescript
And then I ran `npm run start`. If I edit the `src/App.svelte` file the browser refreshes with the changes. If I edit the `src/timer.ts` file, the browser will refresh but there will be no change to the code. Only after restarting the server do I see changes (Or making a change to `src/App.svelte`.

This is what this issue is about correct? If so, what exactly are the workarounds you mentioned in the above comment? It's not clear to me what I need to do to fix this problem. If this problem I'm having is unrelated to this thread, which repo should I submit an issue to?

**Comment by jasonlyu123** [comment]
Because the diagnostics like warning and errors, only get updated when the corresponding svelte file updated. So when any svelte/js file it depended on got updated, It's currently not being updated. I don't know any workaround for the users other than what you mentioned.
The possible solution I mentioned is to just update all of the opened files. I can make a PR later for this.

**Comment by peterreeves** [comment]
Ok, thanks for clarifying. I guess I'll just have to keep an eye on this issue then.

**Comment by dummdidumm** [comment]
@peterreeves you talk about "there is no change to the code". Do you mean the compiled code from `npm run start`? If yes, this is a different issue. This issue is about the IDE diagnostics/intellisense not getting updated in svelte files when there is a change to a js/ts file.

**Comment by peterreeves** [comment]
Yes, there is no change to the compiled code. The browser refreshes but it looks like it receives an identical bundle (the hash is the same). Sounds like this is a separate issue then. I'm having trouble working out where the problem is though. Does my issue sound like a problem with Svelte, the Svelte pre-processor, or rollup (Or something else entirely)? I'm having a lot of trouble trying to work out where this is going wrong.

**Comment by dummdidumm** [comment]
@peterreeves seems like a rollup issue https://github.com/rollup/plugins/issues/478

**Comment by peterreeves** [comment]
Ahh, thanks for pointing me in the right direction.

**Comment by dummdidumm** [comment]
@samdesota could you check with the [nightly version of the extension](https://marketplace.visualstudio.com/items?itemName=svelte.svelte-vscode-nightly) if the issue still occurs to you?

**Comment by samdesota** [comment]
@dummdidumm okay, will checkout later today

**Comment by dummdidumm** [comment]
It's now part of the production release, so you don't have to switch to the nightly anymore to check it.

**Comment by qwabra** [comment]
> It's now part of the production release

https://marketplace.visualstudio.com/items?itemName=svelte.svelte-vscode
v101.2.0

stil haven't !?

**Comment by jasonlyu123** [comment]
It works fine for me 😊. What's the problem you are facing?

**Comment by qwabra** [comment]
Svelte server still need reload for import (from .ts)

_svelte.svelte-vscode-nightly (v99.0.18)_

![2020-07-12-13:04:23](https://user-images.githubusercontent.com/5960180/87243653-6b937b80-c440-11ea-9dee-09698f40d313.png)

**Comment by qwabra** [comment]
VSCode `settings.json`
```json
"files.watcherExclude": {
  "**/*": true,
}
```
is block update

**Comment by jasonlyu123** [comment]
We use file watch to detect js/ts file update.

**Comment by qwabra** [comment]
hmm, `sveltePreprocess` doesn't have watcher?

**Comment by jasonlyu123** [comment]
Build and IDE is separate. And I don't think preprocess needs a watcher

**Comment by dummdidumm** [comment]
Closing as this seems no longer to be an issue. I added a note in the docs about the file watcher issue.
